compositor_render/src/renderer.rs

```diff
@@ -33,7 +34,7 @@ pub mod texture;
 
 pub struct Renderer {
     pub wgpu_ctx: Arc<WgpuCtx>,
-    pub text_renderer_ctx: TextRendererCtx,
+    pub text_renderer_ctx: Mutex<TextRendererCtx>,
```

Comment:
Either remove this mutex or those 2 mutexes inside TextRendererCtx:
- `Mutex<TextRendererCtx>` is safer because you don't need to worry about the order of locking on those mutexes
- Mutexes inside TextRendererCtx could be faster, but we are already processing everything in a single pipeline, so it should be fine.

I would go with the first option

Reply:
Since `text_renderer.prepare` requires mut ref to `font_system` and `swash_cache`, I need to have mut ref access to both fields of `TextRendererCtx` at the same scope. I couldn't find a way to do this having mutex only on `TextRendererCtx`, so I left mutexes on struct fields.